src/components/NavBar/RecentlySearchedAssets.ts

```diff
@@ -0,0 +1,81 @@
+import { Chain, NftCollection, useRecentlySearchedAssetsQuery } from 'graphql/data/__generated__/types-and-hooks'
+import { useAtom } from 'jotai'
+import { atomWithStorage, useAtomValue } from 'jotai/utils'
+import { FungibleToken, GenieCollection, parseFungibleTokens } from 'nft/types'
+import { useCallback, useMemo } from 'react'
+
+type RecentlySearchedAsset = {
+  isNft?: boolean
+  address: string
+  chain: Chain
+}
+
+const recentlySearchedAssetsAtom = atomWithStorage<RecentlySearchedAsset[]>('recentlySearchedAssets', [])
+
+export function useAddRecentlySearchedAsset() {
+  const [searchHistory, updateSearchHistory] = useAtom(recentlySearchedAssetsAtom)
+
+  return useCallback(
+    (asset: RecentlySearchedAsset) => {
+      // Removes the new asset if it was already in the array
+      const newHistory = searchHistory.filter((oldAsset) => oldAsset.address !== asset.address)
+      newHistory.unshift(asset)
+      updateSearchHistory(newHistory)
+    },
+    [searchHistory, updateSearchHistory]
+  )
+}
+
+export function useRecentlySearchedAssets() {
+  const history = useAtomValue(recentlySearchedAssetsAtom)
+  const shortenedHistory = useMemo(() => history.slice(0, 4), [history])
+
+  const { data: queryData, loading } = useRecentlySearchedAssetsQuery({
+    variables: {
+      collectionAddresses: shortenedHistory.filter((asset) => asset.isNft).map((asset) => asset.address),
+      contracts: shortenedHistory
+        .filter((asset) => !asset.isNft)
+        .map((token) => ({
+          address: token.address === 'NATIVE' ? (null as unknown as string) : token.address,
```

Comment:
`(null as unknown as string)` is a temporary measure until the backend accepts "NATIVE" as input. We currently need to pass null for caching to work properly, but our apollo setup doesn't accept null.

Reply:
I'm not exactly sure what's causing it but I'm seeing WETH show up for Popular Tokens instead of ETH
<img width="550" alt="Screen Shot 2023-01-31 at 21 35 13 " src="https://user-images.githubusercontent.com/11512321/215961582-b1f10ae2-b910-40a0-babb-f50b12e4afaf.png">

Reply:
Same thing happens with MATIC on polygon
<img width="559" alt="Screen Shot 2023-01-31 at 21 47 17 " src="https://user-images.githubusercontent.com/11512321/215961923-93caa99d-267a-4d89-b6e4-4e2d39317648.png">

Reply:
great callout, meant to unwrap those in the [migrate_trending](https://github.com/Uniswap/interface/tree/migrate_trending) pr, just fixed on there 👍